code/game/objects/items/devices/PDA/cart.dm

```diff
@@ -500,7 +500,7 @@
 		if(!CbotData.len)
 			CbotData[++CbotData.len] = list("x" = 0, "y" = 0, dir=null, status = null)
 		var/CartData[0]
-		for(var/obj/structure/stool/bed/chair/cart/janicart/B in world)
+		for(var/obj/structure/janitorialcart/B in world)

```

Comment:
Consider replacing all these world checks for the janitorial equipment with a list, like janitorial_equipment.